staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -223,3 +223,13 @@ func prefixFromUserAgent(u string) string {
 	}
 	return buf.String()
 }
+
+var createOptionsGVK = metav1.SchemeGroupVersion.WithKind("CreateOptions")
+
+// createOptionsWithTypeMeta sets the group version kind to meta.k8s.io/v1.CreateOptions.
+func createOptionsWithTypeMeta(options *metav1.CreateOptions) *metav1.CreateOptions {
+	admissionOptions := options.DeepCopy()
+	apiVersion, kind := createOptionsGVK.ToAPIVersionAndKind()
+	admissionOptions.TypeMeta = metav1.TypeMeta{Kind: kind, APIVersion: apiVersion}
```

Comment:
do we still need these `OptionsWithTypeMeta` methods after #77578 is fixed? I remember we talked about this wrapping idea coming from deserialization failure, but TypeMeta is supposed to be optional

Reply:
rather than all the `*OptionsWithTypeMeta` methods, would it make sense to adjust the place where each handler sets up these objects (I count six places) to leave them with `apiVersion: meta.k8s.io/v1` and `kind: ...` set?

Reply:
> do we still need these `OptionsWithTypeMeta` methods after #77578 is fixed? I remember we talked about this wrapping idea coming from deserialization failure, but TypeMeta is supposed to be optional

I've been working on the assumption that although we don't require TypeMeta, that we should provide it do admission webhooks so it's obvious (at a machine readable level) to them which of the various option types they're receiving.

Reply:
> rather than all the `*OptionsWithTypeMeta` methods, would it make sense to adjust the place where each handler sets up these objects (I count six places) to leave them with `apiVersion: meta.k8s.io/v1` and `kind: ...` set?

Yes, that sounds cleaner. I'll update.